File: samples/cs/04primality/prime.cs
using System;

namespace Primality {
    class Exec {
        static bool IsPrime(int number) {
            int limit = (int)Math.Sqrt(number);
            for (int i = 2; i <= limit; i++) {
                if (number % i == 0) {
                    return false;
                }
            }
            return true;
        }

        static void Main(string[] args) {
            if (args.Length < 1) {
                Console.Error.WriteLine("Error: Expected to receive at least 1 argument ./prime <number> [<number> ...]");
                System.Environment.Exit(1);
            }
            foreach (string arg in args) {
                int number = 0;
                try {
                    number = Int32.Parse(arg);
                } catch (System.FormatException) {
                    Console.Error.WriteLine(string.Format("Error: Expected to receive an integer argument. Received {0}.", arg));
                    System.Environment.Exit(2);
                }
                Console.WriteLine(string.Format("{0} => {1}", number, IsPrime(number)));
            }
        }
    }
}

File: samples/java/04primality/prime.java
import java.lang.Math;


public class prime {

    public static boolean isPrime(double number) {
        int limit = (int)Math.sqrt(number);
        for (int i = 2; i <= limit; i++) {
            if (number % i == 0) {
                return false;
            }
        }
        return true;
    }
    
    public static void main(String[] args) {
        if (args.length == 0) {
            System.err.println("Error: Usage 'java prime <number> [<number> ...]'");
            System.exit(1);
        }
        for (String arg : args) {
            int number = 0;
            try {
                number = Integer.parseInt(arg);
            } catch (java.lang.NumberFormatException e) {
                System.err.println(String.format("Error: Expected an integer, received '%s'", arg));
                System.exit(2);
            }
            System.out.println(String.format("%d => %s", number, isPrime(number) ? "true" : "false"));
        }
    }

}
